storb/util/message_signing.py

```diff
@@ -45,7 +46,7 @@ class PieceMessage(BaseModel):
     piece_type: PieceType
 
 
-MessageType = Union[ChunkMessage, TrackerMessage, PieceMessage]
+MessageType = Union[ChunkMessage, TrackerMessage, PieceMessage, Challenge]
```

Comment:
```suggestion
MessageType = ChunkMessage | TrackerMessage | PieceMessage | Challenge
```

You can just do this since we're using Python 3.12